src/Components/Grants/GrantCopy.jsx

```diff
@@ -23,25 +20,36 @@ export default function GrantCopy() {
     history.push(buildOrganizationsLink(`/grants/${grantId}`));
   };
 
-  const handleSubmit = (newGrantFields) => {
-    copyGrant(organizationClient, grantId, newGrantFields)
-      .then((copiedGrant) => {
-        alert("Grant copied!");
-        history.push(buildOrganizationsLink(`/grants/${copiedGrant.id}`));
-      })
-      .catch((error) => {
-        console.error(error);
-        alert(
-          "Eek! Something went wrong when copying the grant. Try again soon."
+  const { mutate: copyGrant } = useMutation(
+    (grantFields) =>
+      GrantsService.copyGrant(organizationClient, grantId, grantFields),
+    {
+      onSuccess: (copyRes) => {
+        alert("Grant created!");
+        history.push(
+          `/organizations/${currentOrganization.id}/grants/${copyRes.id}`
         );
-      });
-  };
+      },
+    }
+  );
+
+  function handleCopyGrant(copyGrantFields) {
+    copyGrant({
+      ...copyGrantFields,
+      title: copyGrantFields.title,
+      fundingOrgId: copyGrantFields.fundingOrgId,
+      rfpUrl: copyGrantFields.rfpUrl,
+      purpose: copyGrantFields.purpose,
+      deadline: copyGrantFields.deadline,
+      totalWordCount: copyGrantFields.totalWordCount,
+    });
```

Comment:
Since all these fields seem to be the same could we rely on the parameter entirely?
```suggestion
    copyGrant(copyGrantFields);
```